activerecord/lib/active_record/autosave_association.rb

```diff
@@ -370,8 +370,14 @@ def validate_collection_association(reflection)
       # enabled records if they're marked_for_destruction? or destroyed.
       def association_valid?(association, record)
         return true if record.destroyed? || (association.options[:autosave] && record.marked_for_destruction?)
-
-        context = validation_context if custom_validation_context?
+        if custom_validation_context?
+          context = validation_context
+        else
+          # If the associated record is unchanged we shouldn't auto validate it.
+          # Even if a record is invalid you should still be able to create new references
+          # to it.
+          return true if !record.new_record? && !record.changed?
```

Comment:
Why doesn't this use `#changed_for_autosave?`

https://api.rubyonrails.org/classes/ActiveRecord/AutosaveAssociation.html#method-i-changed_for_autosave-3F

Reply:
Thank you, @ianterrell !!!